src/pages/Fox/components/FoxTokenBalances.tsx

```diff
@@ -0,0 +1,177 @@
+
+  const relatedAssetIdsFilter = useMemo(() => ({ assetId, onlyConnectedChains: true }), [assetId])
+  const relatedAssetIds = useAppSelector(state =>
+    selectRelatedAssetIds(state, relatedAssetIdsFilter),
+  )
+
+  const accountIdsByChainId = useAppSelector(selectAccountIdsByChainId)
+
+  const assetsWithFoxOnEthereum = useMemo(() => {
+    return [foxAssetId, ...relatedAssetIds]
+  }, [relatedAssetIds])
+
+  const filteredAssets = useMemo(() => {
+    if (selectedFilters.includes(ALL_FILTER_KEY)) return assetsWithFoxOnEthereum
+
+    return assetsWithFoxOnEthereum.filter(assetId => {
+      const chainId = fromAssetId(assetId).chainId
+      return selectedFilters.includes(chainId)
+    })
+  }, [selectedFilters, assetsWithFoxOnEthereum])
+
+  const filters = useMemo<Filter[]>(() => {
+    return [
+      {
+        label: ALL_FILTER_KEY,
+      },
+      ...assetsWithFoxOnEthereum.map(assetId => ({
+        label: chainIdToChainDisplayName(fromAssetId(assetId).chainId),
+        chainId: fromAssetId(assetId).chainId,
+        assetId,
+      })),
+    ]
+  }, [assetsWithFoxOnEthereum])
+
+  const handleFilterClick = (filter: Filter) => {
+    setSelectedFilters(prev => {
+      if (filter.label === ALL_FILTER_KEY) {
+        return [ALL_FILTER_KEY]
+      }
+
+      if (!filter.chainId) return prev
+
+      const newFilters = prev.filter(f => f !== ALL_FILTER_KEY && f !== filter.label)
+
+      if (!prev.includes(filter.chainId)) {
+        newFilters.push(filter.chainId)
+      } else {
+        newFilters.splice(newFilters.indexOf(filter.chainId), 1)
+      }
+
+      return newFilters.length ? newFilters : [ALL_FILTER_KEY]
+    })
+  }
+
+  const filteredAssetsAccountsWithBalances = useMemo(() => {
+    const entries = filteredAssets.reduce<JSX.Element[]>((acc, filteredAssetId) => {
+      if (!selectedAssetAccountId) return acc
+      const filteredAssetChainId = fromAssetId(filteredAssetId).chainId
+
+      if (!filteredAssetChainId) return acc
+      const filteredAssetAccountId = accountIdsByChainId[filteredAssetChainId]
+
+      if (assetAccountNumber === undefined) return acc
+      if (!filteredAssetAccountId?.[assetAccountNumber]) return acc
+
+      acc.push(
+        <AccountEntryRow
+          key={filteredAssetId}
+          accountId={filteredAssetAccountId[assetAccountNumber]}
+          assetId={filteredAssetId}
+          showNetworkIcon={true}
+        />,
+      )
+
+      return acc
+    }, [])
+
+    if (!entries.length) {
+      return <Text translation='foxPage.noAccounts' color='text.subtle' />
+    }
+
+    return <>{entries.map(account => account)}</>
+  }, [accountIdsByChainId, assetAccountNumber, filteredAssets, selectedAssetAccountId])
+
+  return (
+    <Box mb={10}>
+      <ButtonGroup variant='transparent' mb={4} spacing={0}>
+        <HStack spacing={1} p={1} borderRadius='md' {...hstackProps}>
+          {filters.map(filter => {
+            const feeAsset = getFeeAssetByAssetId(assets, filter.assetId)
```

Comment:
preferably-blocking: use `getFeeAssetId()` from adapter by ChainId not to have to `selectAssets` since this is the only usage, or extract me into a component to leverage `selectFeeAssetById` / `selectFeeAssetByChainId`)

Reply:
Done in [e05e4b2](https://github.com/shapeshift/web/pull/7882/commits/e05e4b244591c3a87f0da0fca531273c9a08a4c3)